### PR TITLE
🏃Add baremetal provider in clusterctl pre-defined providers list

### LINKED ISSUE #2356: Add CAPBM to infrastructure provider list in clusterctl
**User Story**

As a user, I would like to use clusterctl (v1alpha3) with the [baremetal provider](https://github.com/metal3-io/cluster-api-provider-baremetal)

**Detailed Description**
I would like to add the baremetal provider in the pre-defined list of providers of clusterctl.

/kind feature

### PR DESCRIPTION
**Pull request by maelk**
**What this PR does / why we need it**:
This PR adds the baremetal infrastructure provider to the clusterctl pre-defined list of providers.

**Which issue(s) this PR fixes**:
Fixes #2356

### CODE REVIEW

**Comment by maelk** [comment]
Thank you for the feedback. I added the missing pieces and fixed the tests. Currently the overview points to a getting-started.md that is WIP. It will be added in CAPBM very soon.

File: docs/book/src/clusterctl/overview.md [hunk 1]
```diff
@@ -151,6 +151,11 @@ For more information about prerequisites, credentials management, or permissions
 
 Please visit the [getting started guide](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/docs/getting-started.md).
 
+{{#/tab }}
+{{#tab Bare Metal}}
```
Comment (fabriziopandini) [inline docs/book/src/clusterctl/overview.md]: I did not check the preview, but I'm pretty sure the new tab should be added to the list of tabs at line 77

Comment (maelk) [reply docs/book/src/clusterctl/overview.md]: fixed, thank you

**Pull request merged**